modules-js/react-fleet/src/form-elements/SelectDropdown.tsx

```diff
@@ -0,0 +1,81 @@
+import React, { SelectHTMLAttributes } from 'react';
+import hash from 'string-hash';
+
+
+interface Props extends SelectHTMLAttributes<HTMLSelectElement> {
+  label: string;
+  error?: string | boolean;
+
+  variant?: 'small';
+  hideLabel?: boolean;
+  hideBlank?: boolean;
+  options: string[];
+}
+
+/**
+ * Stateless dropdown select component.
+ *
+ * Automatically generates an id for the element if one is not provided.
+ *
+ * If hideLabel is specified, the label text will be added as an aria-label
+ * attribute on the select element.
+ *
+ * todo: at narrow widths, padding visually obscures current selection
+ *
+ * todo: currently assumes it will receive an array of strings to create its <option>s; there may be cases when we want an option to have a different value from its displayed text, and pass in an array of objects instead. - 9/21 jm
+ */
+export default function SelectDropdown(props: Props): JSX.Element {
+  const { error, hideLabel, label, required, variant } = props;
+  const id = props.id || `select-${hash(label)}`;
+
+  const classNames = {
```

Comment:
Interesting pattern. It’s nice for documentation, but on the other hand it doesn’t give you an unused variable warning if you stop using one of them.

Reply:
ah, good point.